packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -353,6 +354,24 @@ export class ZipFS extends BasePortableFakeFS {
     return !!this.fds.size;
   }
 
+  async opendirPromise(p: PortablePath, opts?: OpenDirOptions) {
+    return this.opendirSync(p, opts);
+  }
+
+  opendirSync(p: PortablePath, opts: OpenDirOptions = {}): Dir<PortablePath> {
+    const resolvedP = this.resolveFilename(`opendir '${p}'`, p);
+    if (!this.entries.has(resolvedP) && !this.listings.has(resolvedP))
+      throw errors.ENOENT(`opendir '${p}'`);
+
+    const directoryListing = this.listings.get(resolvedP);
+    if (!directoryListing)
+      throw errors.ENOTDIR(`opendir '${p}'`);
+
+    const entries = [...directoryListing];
+
+    return opendir(resolvedP, this, entries);
+  }
```

Comment:
Needs to register an open handle which is closed when `Dir` is closed otherwise ZipOpenFS can close the ZipFS instance while a directory is getting read

Reply:
Thanks for reminding me! Fixed 👍